### PR TITLE
Add user prompt when removing a project. Resolves #90

### LINKED ISSUE #90: Ask user before removing project
When the control to remove a project is clicked, the user should be asked for confirmation first.

### PR DESCRIPTION
**Pull request by jorgenbs**

### CODE REVIEW

File: app/layout.styl [hunk 1]
```diff
@@ -27,7 +27,8 @@ form
   padding-bottom 60px
 
   // TODO: Make responsive
-  width 1170px
+  //width 1170px
+  width: 100%
```
Comment (aknuds1) [inline app/layout.styl]: I don't think this PR should affect the app width?

File: app/routeMap.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ let discourse = require('./discourse')
 
 module.exports = {
   '/': explore,
+  '/explore': explore,
```
Comment (aknuds1) [inline app/routeMap.js]: What's the point of adding this extra route?

File: app/views/project/editProject.js [hunk 3]
```diff
@@ -136,19 +173,9 @@ let EditProjectPad = component('EditProjectPad', (cursor) => {
...
-          ajax.delete(`/api/projects/${project.owner}/${project.projectId}`)
-            .then(() => {
-              logger.debug(`Project successfully deleted '${qualifiedProjectId}'`)
-              router.goTo('/')
-            }, (error) => {
-              logger.warn(`Failed to delete project '${qualifiedProjectId}': ${error}`)
-              // TODO Notify
-              editCursor.set('isWaiting', false)
-            })
+          // TODO: Ask user for confirmations
```
Comment (aknuds1) [inline app/views/project/editProject.js]: What is the TODO for now?

File: app/views/project/editProject.js [hunk 4]
```diff
@@ -57,11 +58,47 @@ let editProject = (cursor) => {
...
+  }
+
+  let closeCallback = () => {
+    cursor.set('showDeleteProjectDialog', false)
+  }
+  let title = 'Delete'
+  let message = 'Are you sure you want to delete this project?'
+
+  return notification.question(title, message, yesCallback, closeCallback)
+
```
Comment (aknuds1) [inline app/views/project/editProject.js]: Why the extra newline before closing the function?

File: app/views/project/editProject.js [hunk 5]
```diff
@@ -57,11 +58,47 @@ let editProject = (cursor) => {
...
+  return notification.question(title, message, yesCallback, closeCallback)
+  
+})
+
 let EditProjectPad = component('EditProjectPad', (cursor) => {
   let editCursor = cursor.cursor('editProject')
   let projectCursor = editCursor.cursor('project')
   let project = projectCursor.toJS()
+  let showDialog = cursor.get('showDeleteProjectDialog') === true
```
Comment (aknuds1) [inline app/views/project/editProject.js]: I think the `=== true` can be dropped. If it has to be a boolean value, rather than truthy, the typical idiom is to use `!!`.

File: app/views/project/editProject.js [hunk 6]
```diff
@@ -57,11 +58,47 @@ let editProject = (cursor) => {
...
+    
+    ajax.delete(`/api/projects/${project.owner}/${project.projectId}`)
+      .then(() => {
+        logger.debug(`Project successfully deleted '${qualifiedProjectId}'`)
+        router.goTo('/')
+      }, (error) => {
+        logger.warn(`Failed to delete project '${qualifiedProjectId}': ${error}`)
+      })
+      .finally(() => {
+        cursor.set('showDeleteProjectDialog', false)
```
Comment (aknuds1) [inline app/views/project/editProject.js]: I think it would be better to update all of the cursor state at once, as this may be buggy due to cursors being immutable.

File: app/views/project/editProject.js [hunk 7]
```diff
@@ -57,11 +58,47 @@ let editProject = (cursor) => {
...
+        editCursor.set('isWaiting', false)
```
Comment (aknuds1) [inline app/views/project/editProject.js]: See above note.

**Comment by aknuds1** [comment]
I suspect your non-related changes are due to the PR being from `master`? I make the mistake of using master for PRs sometimes too, and end up submitting unintended changes as a result.

File: app/routeMap.js (hunk 2 above)
Comment (jorgenbs) [inline app/routeMap.js]: In case root view changes away from explore (say it goes to a splash screen instead). 
So that after a user deletes a project I can router.goTo('/explore') instead of goTo('/'), which would create a weird flow if '/' no longer points to explore.

Comment (aknuds1) [inline app/routeMap.js]: Let's postpone this until we need it? For consistency, I suspect we should redirect / to /explore if we really want that route. Otherwise, it's a bit of a pain.

File: app/layout.styl (hunk 1 above)
Comment (jorgenbs) [inline app/layout.styl]: You are right, its more a general improvement. I just added it because loader was unadjusted when state is isWaiting.

File: app/views/project/editProject.js (hunk 3 above)
Comment (jorgenbs) [inline app/views/project/editProject.js]: removed

File: app/views/project/editProject.js (hunk 4 above)
Comment (jorgenbs) [inline app/views/project/editProject.js]: Just a personal opinion about readability and breathing room. A lot of the source code has very little breathing room, and I guess I just felt compelled to add more space. I added a newline on the top of the function too.

I can remove the bottom space, but take the code below for example. Its quite intimidatingly compact.

``` javascript
let editProject = (cursor) => {
  let editCursor = cursor.cursor('editProject')
  let editProject = editCursor.toJS()
  uploadProject(editProject.project, editCursor, cursor)
    .then(({title, description, instructions, tags, licenseId, username, pictureFiles, files,}) => {
      logger.debug(`Picture files:`, pictureFiles)
      logger.debug(`Files:`, files)
      logger.debug(`title: ${title}, description: ${description}, tags: ${S.join(`,`, tags)}`)
      let qualifiedProjectId = `${editProject.owner}/${editProject.projectId}`
      let data = {
        title,
        description,
        instructions,
        tags,
        licenseId,
        pictures: pictureFiles,
        files,
      }
      logger.debug(`Updating project '${qualifiedProjectId}'...:`, data)
      ajax.putJson(`/api/projects/${qualifiedProjectId}`, data)
```

Compared to further up in the function I wrote

``` javascript
let DeleteProjectDialog = component('DeleteProjectDialog', (cursor) => {

  let yesCallback = () => {
    let editCursor = cursor.cursor('editProject')
    let projectCursor = editCursor.cursor('project')
    let project = projectCursor.toJS()
    let qualifiedProjectId = `${project.owner}/${project.projectId}`

    editCursor.set('isWaiting', true)

    ajax.delete(`/api/projects/${project.owner}/${project.projectId}`)
```

Comment (aknuds1) [inline app/views/project/editProject.js]: I tend to like compact code, there is a difference between unnecessary whitespace and delimiting whitespace IMO. By the latter I mean whitespace that delimits logical blocks of code. Whitespace at the beginning and especially the end of functions just read like noise to me. I always ask myself whether that was left on purpose or not, because it just looks like it was left behind by accident.

File: app/views/project/editProject.js (hunk 5 above)
Comment (jorgenbs) [inline app/views/project/editProject.js]: Added `!!`

File: app/layout.styl [hunk 8]
```diff
@@ -27,7 +27,7 @@ form
   padding-bottom 60px
 
   // TODO: Make responsive
-  width 1170px
+  width: 100%
```
Comment (aknuds1) [inline app/layout.styl]: Can you remove this change?

File: app/views/project/editProject.js [hunk 9]
```diff
@@ -51,17 +53,54 @@ let editProject = (cursor) => {
...
+        logger.debug(`Project successfully deleted '${qualifiedProjectId}'`)
+        router.goTo('/explore')
       }, (error) => {
-        logger.warn(`Uploading files/pictures failed: ${error}`, error.stack)
-        editCursor = editCursor.set('isWaiting', false)
+        logger.warn(`Failed to delete project '${qualifiedProjectId}': ${error}`)
       })
-}
+      .finally(() => {
+        editCursor.mergeDeep({
```
Comment (aknuds1) [inline app/views/project/editProject.js:80]: Looks good to me, thanks.

File: app/layout.styl (hunk 8 above)
Comment (jorgenbs) [inline app/layout.styl]: should I put it in another PR then?

File: app/layout.styl [hunk 10]
```diff
@@ -25,9 +25,7 @@ form
 
 #content
   padding-bottom 60px
-
-  // TODO: Make responsive
-  width 1170px
+  width: 100%
```
Comment (aknuds1) [inline app/layout.styl]: Why add as part of this PR? I think I would rather discuss this change later, as 100% isn't necessarily the right look.

Comment (jorgenbs) [inline app/layout.styl]: I tried to make it a part of issue #99. Reverted. Take the discussion there
